synapse/storage/databases/main/events.py

```diff
@@ -1954,19 +1954,19 @@ def _handle_redaction(self, txn, redacted_event_id):
         )
 
     def _store_room_topic_txn(self, txn, event):
-        if hasattr(event, "content") and "topic" in event.content:
+        if hasattr(event, "content") and isinstance(event.content.get("topic"), str):
```

Comment:
Do we need any sort of background update to excise previously bad data or anything?

Reply:
That sounds like a better idea.